### PR TITLE
JJ50 v2 Support

### PR DESCRIPTION
**Pull request by stephen776**
A new "V2" version of the JJ50 PCB is being sold by KPRepublic that uses ATMEGA32U4 chip. This PR adds a "rev2" version for the JJ50 that works with this updated PCB.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/jj50/rev2/keyboard.json [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+            "rgb_test": true,
+            "alternating": true,
+            "twinkle": true
+        }
+    },
+    "ws2812": {
+        "driver": "i2c"
+    },
+    "processor": "atmega32u4",
+    "bootloader": "bootloadhid",
```
[the remark this answers is not in the document]
Comment (stephen776) [reply keyboards/kprepublic/jj50/rev2/keyboard.json]: ahh yes you are correct. Worked with toolbox but not CLI. I have updated to use `atmel-dfu` and that work via CLI

File: keyboards/kprepublic/jj50/rev2/keyboard.json [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            "D7",
+            "D6",
+            "D4"
+        ],
+        "rows": ["D3", "D5", "E6", "F1", "F4"]
+    },
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "mousekey": false,
```
Comment (waffle87) [inline keyboards/kprepublic/jj50/rev2/keyboard.json:31]: This is generally preferred to be enabled by default. If firmware size is a concern, you can disable some of the RGB effects, like `rgb_test`.
```suggestion
        "mousekey": true,
```

Comment (stephen776) [reply keyboards/kprepublic/jj50/rev2/keyboard.json]: Updated. firmware size is still good 👍

**Review by waffle87 — APPROVED** [review]

File: keyboards/kprepublic/jj50/rev2/keyboard.json
Comment (fauxpark) [inline keyboards/kprepublic/jj50/rev2/keyboard.json:1]: Please run `qmk format-json -i` on this file.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/jj50/rev2/keyboard.json
Comment (stephen776) [reply keyboards/kprepublic/jj50/rev2/keyboard.json:1]: done

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**